File: app/schemas/job.py
"""
Esquemas de Respuesta para Jobs y sus Pasos (Job & JobStep Response).
Permiten devolver informacion detallada del estado de ejecucion.
Organizan los datos de forma anidada (Un Job contiene una lista de Steps).
"""
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.schemas.job_result import JobResultResponse

class JobStepResponse(BaseModel):
    # Datos detallados de cada paso individual del trabajo
    id: UUID
    job_id: UUID
    step_order: int      # Posicion en la secuencia de ejecucion
    name: str            # Nombre del paso
    step_type: str       # Tipo tecnico del paso
    connector_type: str | None # Conector usado (opcional)
    input_json: dict     # Parametros de entrada en formato JSON
    status: str          # Estado (pending, success, failed, etc.)
    created_at: datetime
    started_at: datetime | None
    finished_at: datetime | None

    model_config = ConfigDict(from_attributes=True)

class JobResponse(BaseModel):
    # Datos principales del trabajo (Job)
    id: UUID
    command_id: UUID
    status: str
    intent_name: str
    risk_level: str
    created_at: datetime
    started_at: datetime | None
    finished_at: datetime | None
    
    # Lista anidada: Aqui se incluyen todos los JobStepResponse asociados
    # Si el Job no tiene pasos, devuelve una lista vacia por defecto []
    steps: list[JobStepResponse] = []

    model_config = ConfigDict(
        # Permite convertir objetos de SQLAlchemy (modelos) a estos esquemas JSON
        from_attributes=True
    )

File: frontend/src/types/job.ts
export type JobStatus =
  | "created"
  | "ready_to_execute"
  | "queued"
  | "running"
  | "completed"
  | "failed"
  | "approval_pending"
  | "rejected";

export type JobStepStatus =
  | "pending"
  | "running"
  | "completed"
  | "failed"
  | "blocked";

export type JobRiskLevel = "low" | "medium" | "high" | "unknown";

export type JobStep = {
  id: string;
  job_id: string;
  step_order: number;
  name: string;
  step_type: string;
  connector_type: string | null;
  input_json: Record<string, unknown>;
  status: JobStepStatus;
  created_at: string;
  started_at: string | null;
  finished_at: string | null;
};

export type JobResult = {
  id: string;
  job_id: string;
  success: boolean;
  summary: string | null;
  result_json: Record<string, unknown> | null;
  created_at: string;
};

export type Job = {
  id: string;
  command_id: string;
  status: JobStatus;
  intent_name: string;
  risk_level: JobRiskLevel;
  created_at: string;
  started_at: string | null;
  finished_at: string | null;
  steps: JobStep[];
  job_result: JobResult | null;
};